src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.exception.NoRequiredAnnotationException;
+import core.basesyntax.factory.BetDaoFactory;
+import core.basesyntax.factory.UserDaoFactory;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class Injector {
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                BetDao betDao = BetDaoFactory.getBetDao();
+                if (field.getType().equals(BetDao.class)
+                        && betDao.getClass().isAnnotationPresent(Dao.class)) {
+                    field.setAccessible(true);
+                    field.set(instance, betDao);
+                } else {
+                    UserDao userDao = UserDaoFactory.getUserDao();
+                    if (field.getType().equals(UserDao.class)
+                            && userDao.getClass().isAnnotationPresent(Dao.class)) {
+                        field.setAccessible(true);
+                        field.set(instance, userDao);
+                    } else {
+                        throw new NoRequiredAnnotationException(String.format(
+                                "The %s class has no @Dao annotation", field.getType().getName()));
+                    }
+                }
```

Comment:
I would recommend to simplify it and change the name of your custom Exception
```suggestion
                BetDao betDao = BetDaoFactory.getBetDao();
                UserDao userDao = UserDaoFactory.getUserDao();
                field.setAccessible(true);
                if (field.getType().equals(BetDao.class)
                        && betDao.getClass().isAnnotationPresent(Dao.class)) {
                    field.set(instance, betDao);
                } else if (field.getType().equals(UserDao.class)
                            && userDao.getClass().isAnnotationPresent(Dao.class)) {
                    field.set(instance, userDao);
                } else {
                    throw new NoImplementationException(field.getName()
                            + " doesn't have an implementation that can be injected");
                }
```

Reply:
done